presto-jdbc/src/main/java/com/facebook/presto/jdbc/ConnectionParameters.java

```diff
@@ -34,15 +33,15 @@
 {
     private static final String JDBC_URL_START = "jdbc:";
 
-    private final AtomicReference<String> catalog = new AtomicReference<>();
-    private final AtomicReference<String> schema = new AtomicReference<>();
-
     private static final Splitter QUERY_SPLITTER = Splitter.on('&').omitEmptyStrings();
     private static final Splitter ARG_SPLITTER = Splitter.on('=').limit(2);
 
     private final HostAndPort address;
     private final URI uri;
 
+    private String catalog;
```

Comment:
`catalog `and `schema` are parsed within the method `initCatalogAndSchema` called from constructor. Therefore, there is no simple way to set both values in constructor.